clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -93,25 +93,36 @@
     private final String metricGrpPrefix;
     private final Map<String, String> metricTags;
     private final ChannelBuilder channelBuilder;
-    private final Map<String, Long> lruConnections;
-    private final long connectionsMaxIdleNanos;
     private final int maxReceiveSize;
     private final boolean metricsPerConnection;
-    private long currentTimeNanos;
-    private long nextIdleCloseCheckTime;
-
+    private final IdleExpiryManager idleExpiryManager;
 
     /**
      * Create a new nioSelector
+     *
+     * @param maxReceiveSize Max size in bytes of a single network receive
+     * @param connectionMaxIdleMs Max idle connection time (negative values disable idle timeout)
```

Comment:
Sure. I also considered creating a Builder for Selector, but it seemed like overkill.